packetbeat/pb/event.go

```diff
@@ -416,8 +416,17 @@ func marshalStruct(m mapstr.M, key string, val reflect.Value) error {
 	}
 
```

Comment:
On L412, can you please add an assertion that the given value is a struct to prevent a panic. This is just to be safe w.r.t. to the comment you made "assume a pointer is a struct".

```go
	if val.Type().Kind() != reflect.Struct {
		return fmt.Errorf("value must be a struct or a pointer to a struct, but got %v at key %s", val.Type(), key)
	}
```